bin/main.rb

```diff
@@ -1,85 +1,102 @@
-      l_0, l_1, l_2 = l[0].to_i, l[1].to_i, l[2].to_i 
-        return true if @board[l_0] == @board[l_1] && @board[l_1] == @board[l_2]
-      
-    end
-    false
-  end
-
-end
+require './lib/tic_tac_toe'
+require './lib/player'
 
 def start_game
   puts "Welcome to Tic Tac Toe game.\n\n"
-  game = TicTacToe.new(ask_names)
-  print_board(game)
+  game = TicTacToe.new(get_player_names)
+  print_game_board(game)
 
   turn(game) until game.game_over?
 end
 
-# ask the player names
-def ask_names
+def get_player_names
+  accept_players = nil
   players = []
-  2.times do |i|
-    print "Please tell us player #{i + 1} name: "
-    name = gets.chomp
-    # TODO: validate (min 3 letters)
-    players[i] = Player.new(i, name)
+  until accept_players == true
+    2.times do |i|
+      accept_name = nil
+      until accept_name == true
+        print "Please tell us player #{i + 1} name: "
+        name = gets.chomp
+        name = name.gsub(/\d/, "")   
+        name = name.gsub(/\W/, "")  
+        if name.length < 3
+          puts 'Please enter at least 3 A-Z-only characters'
+        else 
+          accept_name = true
+        end
+      end
+      players[i] = Player.new(i, name)
+    end
+    if players[0].name.downcase.eql?(players[1].name.downcase)
+      puts 'Players cant have the same name!'
+    else
+      accept_players = true
+    end
   end
   players
 end
 
-# Print the keyboard with numbers from 1 to 9
-def print_board(game)
-  puts "#{game.board[0..2].join(" ")}"
-  puts "#{game.board[3..5].join(" ")}"
-  puts "#{game.board[6..8].join(" ")}" 
+def print_game_board(game)
+  puts "\n"
+  puts "#{game.board[0..2].join(" | ")}"
+  puts "#{game.board[3..5].join(" | ")}"
+  puts "#{game.board[6..8].join(" | ")}"
+  puts "\n"
 end
 
 def turn(game)
-  print game.players[game.turn % 2].name + ' please choose a number: '
-  num = gets.chomp
-  puts num
+  # TODO: validate (only number, min 1 && max 1, available number)
+  # http://ruby.bastardsbook.com/chapters/exception-handling/
 
-  game.set(num.to_i)
+  input = ''
 
-  # if num >= 1 && num <= 9) do
-  #   puts 'Number already choosen';
-  # end
+  loop do
+    err = nil
 
-  game.turn += 1
-  print_board(game)
+    print game.current_player.name + ' please choose an unused digit: '
+    input = gets.chomp
+
+    if input.length != 1 || /\D/.match(input)
+      err = "\n\n #{input} is not a digit \n\n" 
+      puts err
+    else
+      unless game.board.include?(input.to_i)
+        err = "\n\n number #{input} was already used \n\n"
+        puts err
+      end
+    end
+
+    break unless err
+  end
+
+  game.set(input.to_i)
+  print_game_board(game)
+  end_game(game) if game.game_over?
+  
+end
+
+def end_game(game)
+  if game.status.eql? game.enum_status[:win]
+    puts "Congratulation #{game.current_player.name} you win!"
+  else
+    puts "We have a draw!"
+  end
+
+  puts "\nWant to play again? [Y/n]"
+  play_again = gets.chomp
+
+  play_again.eql?('n') ? finish : start_game
```

Comment:
[OPTIONAL]
You are checking if the is `n` otherwise you start game. Is this best? what if person typed no and not `Y`?
Maybe be more specific. If it's neither `Y` or `n` prompt them to type answer again.

See below image for context:
![Screen Shot 2019-10-12 at 7 34 13 PM](https://user-images.githubusercontent.com/31193823/66708944-be140000-ed27-11e9-95b7-8700bf68fe35.png)